packages/wallet/src/redux/reducers/funding.ts

```diff
@@ -154,10 +183,7 @@ const aSubmitDeployToMetaMaskReducer = (state: states.ASubmitDeployInMetaMask, a
     case actions.TRANSACTION_SUBMITTED:
       return states.waitForDeployConfirmation({ ...state, transactionHash: action.transactionHash });
     case actions.TRANSACTION_SUBMISSION_FAILED:
-      return states.waitForChannel({
-        ...state,
-        messageOutbox: fundingFailure(state.channelId, "Other", action.error.message),
-      });
+      return states.deployTransactionFailed(state);
```

Comment:
@lalexgap Checking to see if my understanding is correct here. We are currently not handling the case where transaction submission succeeds, but the transaction is rejected for some reason after the submission.

Reply:
@kerzhner That's correct. We can probably extend this to cover where a transaction errors out after submission as well. (Although retrying a failed transaction might not help a lot?)